docs/json.adoc

```diff
@@ -0,0 +1,156 @@
+:page-title: JSON
+:page-description: Extends JUnit Jupiter with `@JsonFileSource`, a parametrized test that creates test based on a JSON Source
+
+The `@JsonFileSource` annotation lets you use JSON files from the classpath or local file system to create parametrized tests.
+
+By default the root of the source will be treated as candidate for the test arguments.
+If the root is an object then the entire object will be one argument, if the root is an array then every element of the array will be one argument.
+
+There is also `JsonFileSource#data` that can be used to tell the extraction mechanism to use the element with that name to look for the source of the data.
+
+Depending on how the test method parameters the extraction of the values might differ.
+
+== Single Argument Methods
+
+When the method has a single argument then the json object argument will be converted to that type.
+
+.Argument type
+[source,java]
+----
+public class Customer {
+	public String name;
+	public String height;
+
+	@Override
+    public String toString() {
+        return "Customer{" + "name='" + name + '\'' + ", height=" + height + '}';
+    }
+}
+----
+
+.JSON Source
+[source,json]
+----
+[
+  {
+    "name": "Luke",
+    "height": 172
+  },
+  {
+    "name": "Yoda",
+    "height": 66
+  }
+]
+----
+
+[source,java]
+----
+@ParametrizedTest
+@JsonFileSource(resources = "/customers.json")
+void singleCustomer(Customer customer) {
+    assertThat(Collections.singleton(tuple(customer.getName(), customer.getHeight())))
+            .containsAnyOf(
+                tuple("Luke", 172),
+                tuple("Yoda", 66)
+            );
```

Comment:
I appreciate the attention to detail, but I don't think we need to flesh out the test. 😉

Reply:
😄  I'll remove them from the docs